### ISSUE TITLE
settingUpdate incorrect definition

### ISSUE THREAD

**Issue by pinthenet** [KillerBOSS2019/TouchPortal-API#3]
First, thanks for the API - it makes things much easier. I had quite a struggle trying to get setting updates to work until I noticed after looking through the code that  
`@TPClient.on('settingUpdate') # This Function will get called Everytime when someone changes something in your plugin settings
def Settings(client, data):
    print('received data from settings!')`

should actually be 
`@TPClient.on('settings') # This Function will get called Everytime when someone changes something in your plugin settings
def Settings(client, data):
    print('received data from settings!')`

based on `TYPES.onSettingUpdate - "settings"`
It works now :-D

Also, 'closePlugin' doesn't seem to work, but I read on Discord that this is apparently broken in Touch Portal in the latest version?

**Comment by KillerBOSS2019** [comment]
Ahh I see thanks with the closePlugin part is bit odd other sdks such as C, and node.js when you close the TouchPortal fully it will not send the closePlugin so worked around when so now it dectects if the Socket is closed or not if it is it will just send {"type":"closePlugin"}

**Comment by KillerBOSS2019** [comment]
Does the closePlugin works in any way? Extra notes if you are stuggling you can check out my other plugins like Win-Mover I think that's the only one uses the sdk

**Comment by KillerBOSS2019** [comment]
Fixed the README.md from "settingUpdate" to "settings"

**Comment by KillerBOSS2019** [comment]
I haven't update the code on github yet.. but it is updated on the pypi.org so if your using the file in github the closePlugin will not work so you should use pypi.org just a heads up will update asap

**Comment by pinthenet** [comment]
Thanks for the updates. I worked around the closePlugin 'problem' by quitting if the TouchPortalServices.exe is not running. A bit 'quick & dirty' (cheap & cheerful) but it works for now. I'll update from  pypi version and see how it works. I read a post on Discord that suggested that Touch Portal itself doesn't send the disconnect with the latest and possibly previous versions.

**Comment by pinthenet** [comment]
FYI – I updated from pypi and I now get a closePlugin callback😃- congrats.  I had to keep my process check though as I can't get the plugin to exit otherwise… I put an exit() after the disconnect, and a check in my 500ms loop, but the plugin continuesrunning, even though the parent process and all other child processes have closed.

 

I find this in the Touch Portal log (I saw this before your update too)

 

20:30:45.232 - [LOG] (Services) Closing the windows services...

20:30:45.233 - [LOG] (Services) Force close the service process...

20:30:45.241 - [LOG] (Services) Force closed received, shutting down the service

20:30:46.736 - [ERROR] (Plugin System) Server - Socket exits: socket closed | socket closed

20:30:46.737 - [ERROR] (Plugin System) Error: Shutting down the plugin communication service... Restart Touch Portal!

 

From: John Payne ***@***.***> 
Sent: 13 April 2021 20:16
To: 'KillerBOSS2019/TouchPortal-API' ***@***.***>
Subject: RE: [KillerBOSS2019/TouchPortal-API] settingUpdate incorrectdefinition (#3)

 

Thanks for the updates. I worked around the closePlugin 'problem' by quitting if the TouchPortalServices.exe is not running. A bit 'quick & dirty' (cheap & cheerful) but it works for now. I'll update from  pypi version and see how it works. I read a post on Discord that suggested that Touch Portal itself doesn't send the disconnect with the latest and possibly previous versions.
